flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java

```diff
@@ -183,6 +209,41 @@ public void notifyCheckpointComplete(long checkpointId, OperatorChain<?, ?> oper
 		env.getTaskStateManager().notifyCheckpointComplete(checkpointId);
 	}
 
+	@Override
+	public void notifyCheckpointAborted(long checkpointId, OperatorChain<?, ?> operatorChain, Supplier<Boolean> isRunning) throws Exception {
+
+		if (isRunning.get()) {
+			LOG.debug("Notification of aborted checkpoint for task {}", taskName);
+			// only happens when the task always received checkpoints to abort but never trigger or executing.
+			if (abortedCheckpointIds.size() >= DEFAULT_MAX_RECORDED_ABOTRED_CHECKPOINTS) {
+				abortedCheckpointIds.pollFirst();
+			}
+
+			boolean canceled = closeableRegistry.cancelAsyncCheckpointRunnable(checkpointId);
```

Comment:
I think the `channelWrittenFuture` is not canceled is during normal cancel phase existed even this feature is not introduced. And storing the `Future<?>` cannot ensure canceling that future could discard those written state handles, thus we should store the `AsyncCheckpointRunnable` itself.

Reply:
`channelWrittenFuture` is canceled by the `ChannelStateWriter` during "normal" abort (its state handle doesn't need to be discarded). 
This PR only "speeds up" the abort, cancellation marker will follow, right?  So eventually it will be canceled too.

Any other interruptible operation (e.g. reporting to JM) should be interrupted by canceling the future returned by the `executorService`.  But this is probably out of scope of this PR.

Reply:
Thanks for your clarification, I would also try to call `ChannelStateWriter#abort` during `SubtaskCheckpointCoordinator#notifyCheckpointAborted` and add logic to verify this in unit test.